File: CursoPOOUber/PHP/car.php
<?php

require_once('account.php');
class Car {

    public $id;
    public $license;
    public $driver;
    protected $passenger; // Es importante que esta propiedad esté marcada como protected

    public function __construct($license, $driver){
        $this->license = $license;
        $this->driver = $driver;
    }

    public function printDataCar() {
        echo "Licencia: $this->license Driver: ".$this->driver->name;
    }
    
}
?>

File: CursoPOOUber/JAVA/Car.java
class Car {
    Integer id;
    Integer passegenger;
    String license;
    Account driver;
    /*String driver;*/
    /* metodo constructor de java, Car es el nombre de la clase, dentro de los parentesis van los parametros y además de esto, el tipo de parametro o dato
    ejemplo: String license, 
    además de eso hacemos el metodo de herencia en la clase, cambiamos el String driver, y heredamos Account driver que es la clase ya definida
    */
    public Car(String, license, Account diver){
        this.license = license;
        this.driver = driver;
    }
    void printDataCar() {
        System.out.println("License: " + license + "Driver: " + driver.name);
    }
}
